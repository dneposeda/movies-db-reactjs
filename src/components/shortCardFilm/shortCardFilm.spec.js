import React from 'react';
import ShortCartFilm from './ShortCardFilm';
import { mount } from 'enzyme';

const mockData = {
    genre: "Oscar winning Movie",
    id: 2,
    img: "https://image.tmdb.org/t/p/w500/3kcEGnYBHDeqmdYf8ZRbKdfmlUy.jpg",
    title: "Reservoir Reservoir",
    year: 2019,
}

describe('ShortCartFilm', () => {
    it('renders correctly', () => {
        const component = mount (
            <ShortCartFilm film={ mockData }/>
        );
        const title = component.find('.card-film__title')

        expect(component).toMatchSnapshot();
        expect(component.find('.card-film')).toHaveLength(1);
        expect(title.text()).toEqual(mockData.title);
    });
});
