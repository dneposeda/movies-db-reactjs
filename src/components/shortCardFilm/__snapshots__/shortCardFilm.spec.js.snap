// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShortCartFilm renders correctly 1`] = `
<ShortCartFilm
  film={
    Object {
      "genre": "Oscar winning Movie",
      "id": 2,
      "img": "https://image.tmdb.org/t/p/w500/3kcEGnYBHDeqmdYf8ZRbKdfmlUy.jpg",
      "title": "Reservoir Reservoir",
      "year": 2019,
    }
  }
>
  <div
    className="col-lg-4 col-md-6 col-sm-6"
  >
    <div
      className="card-film"
    >
      <div
        className="card-film__cover"
      >
        <img
          alt="Reservoir Reservoir"
          className="card-film__img"
          src="https://image.tmdb.org/t/p/w500/3kcEGnYBHDeqmdYf8ZRbKdfmlUy.jpg"
        />
      </div>
      <div
        className="card-film__info"
      >
        <div
          className="card-film__title"
        >
          Reservoir Reservoir
        </div>
        <div
          className="card-film__year"
        >
          2019
        </div>
      </div>
      <div
        className="card-film__genre"
      >
        Oscar winning Movie
      </div>
    </div>
  </div>
</ShortCartFilm>
`;
